src/libraries/Microsoft.Extensions.Logging.Abstractions/src/System/Collections/Generic/ValueListBuilder.Pop.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// These public methods are required by RegexWriter.
+    /// </summary>
+    internal ref partial struct ValueListBuilder<T>
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public T Pop()
+        {
+            _pos--;
+            return _span[_pos];
+        }
+    }
+}
```

Comment:
note: i'm copying what the regex generator does.  it's unclear to me why `Pop` is not just in the linked ValueListBuilder file that is referenced.